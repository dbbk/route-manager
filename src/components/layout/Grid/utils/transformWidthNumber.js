export default function transformWidthNumber(w: number) {
  return w >= 0 && w <= 1 ? `${w * 100}%` : `${w}px`;
}
