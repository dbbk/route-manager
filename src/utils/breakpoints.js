// @flow

// Breakpoints are defined as a map of (name: minimum width)
// https://zpl.io/25RJPjV
const breakpoints = {
  xs: 0, // officially we support a minimum width of 320px
  sm: 569,
  md: 769,
  lg: 961,
  xlg: 1221, // DEPRECATED
};

export default breakpoints;
