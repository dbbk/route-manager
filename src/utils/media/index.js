// @flow

import bp from 'src/utils/breakpoints';

export default function media(label: string) {
  return () => {
    switch (label) {
      case 'xs':
        return `@media (max-width: ${bp.sm - 1}px)`;

      case 'sm':
        return `@media (min-width: ${bp.sm}px) and (max-width: ${bp.md - 1}px)`;

      case 'md':
        return `@media (min-width: ${bp.md}px) and (max-width: ${bp.lg - 1}px)`;

      case 'lg':
        return `@media (min-width: ${bp.lg}px)`;

      case 'to-sm':
        return `@media (max-width: ${bp.sm - 1}px)`;

      case 'to-md':
        return `@media (max-width: ${bp.md - 1}px)`;

      case 'to-lg':
        return `@media (max-width: ${bp.lg - 1}px)`;

      case 'from-sm':
        return `@media (min-width: ${bp.sm}px)`;

      case 'from-md':
        return `@media (min-width: ${bp.md}px)`;

      case 'from-lg':
        return `@media (min-width: ${bp.lg}px)`;

      default:
        return null;
    }
  };
}
