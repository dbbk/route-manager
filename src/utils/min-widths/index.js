// @flow

// minWidths defines the minimum widths of our responsive breakpoints
// as a map of (name: minimum width)
// https://zpl.io/25RJPjV
const minWidths = {
  xs: 0, // officially we support a minimum width of 320px
  sm: 569,
  md: 769,
  lg: 961,
  xlg: 1221, // DEPRECATED
};

export default minWidths;
