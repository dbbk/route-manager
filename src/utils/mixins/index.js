// @flow

import { css } from 'styled-components';
import { prop, switchProp, withProp } from 'styled-tools';
import { darken, lighten } from 'polished';

export const colorMixin = switchProp('color', {
  black: css`
    ${prop('theme.colors.greyscale.black')};
  `,
  grey: css`
    ${prop('theme.colors.greyscale.dark')};
  `,
  white: css`
    ${prop('theme.colors.white')};
  `,
  red: css`
    ${prop('theme.colors.danger')};
  `,
  primary: css`
    ${prop('theme.colors.brand.primary')};
  `,
  inherit: 'inherit',
});

// NOTE while we are lighening and darkening colours using withProp,
// "babel-plugin-polished" will not work.
export const hoverColorMixin = switchProp('color', {
  black: css`
    ${withProp('theme.colors.greyscale.black', lighten(0.1))};
  `,
  grey: css`
    ${withProp('theme.colors.greyscale.dark', darken(0.1))};
  `,
  white: css`
    ${withProp('theme.colors.white', darken(0.1))};
  `,
  red: css`
    ${withProp('theme.colors.danger', darken(0.1))};
  `,
  primary: css`
    ${withProp('theme.colors.brand.primary', darken(0.1))};
  `,
  inherit: 'inherit',
});
