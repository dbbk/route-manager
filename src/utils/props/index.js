// @flow

export function blacklistProps(props: {}, blacklist: Array<string> = []): {} {
  const newProps = { ...props };

  blacklist.forEach(blacklistedProp => delete newProps[blacklistedProp]);

  return newProps;
}
