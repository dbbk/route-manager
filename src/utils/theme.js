// @flow
import { darken } from 'polished';

export function hoverColor(color: string): string {
  return darken(0.05, color);
}

export default {
  colors: {
    white: 'white',
    success: '#57C0B4',
    danger: '#DE5E5E',
    dangerBackground: '#FDF6F6', // primary with transrency 5% on white background
    info: '#A3A6CB',
    star: '#FFCD00',
    background: {
      secondary: '#F9F9FC',
    },
    greyscale: {
      black: '#242529',
      dark: '#A2A4AD',
      silver: '#E2E3E6',
      grey: '#F5F5F5',
    },
    brand: {
      primary: '#8C64CB',
      primaryBackground: '#F9F7FC', // primary with transrency 5% on white background
    },
    facebook: {
      primary: '#4B69AC',
    },
    shadowColor: 'rgba(36, 37, 41, 0.15)',
  },
  animationDuration: {
    shortest: '50ms',
    short: '350ms',
  },
  fontFamily: 'brandon-text, Helvetica Neue, Helvetica, Arial, sans-serif',
  fontWeight: {
    light: '300',
    regular: '400',
    bold: '700',
  },
  breakpoints: {
    xs: 0,
    sm: 568,
    md: 768,
    lg: 960,
    xlg: 1220,
  },
  clipPath: {
    polygonBottomLarge: 'polygon(0 0, 100% 0, 100% 100%, 0 85%)',
  },
};
