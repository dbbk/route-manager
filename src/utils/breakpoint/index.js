/* eslint consistent-return: 0 default-case: 0 */
/* stylelint-disable */

import { css } from 'styled-components';
import theme from '../theme';

export default function breakpoint(label: string) {
  return (...args: any) => () => {
    switch (label) {
      case 'xs':
        return css`
          @media (max-width: ${theme.breakpoints.xs}px) {
            ${css(...args)};
          }
        `;

      case 'sm':
        return css`
          @media (min-width: ${theme.breakpoints.xs}px) and (max-width: ${theme
              .breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'md':
        return css`
          @media (min-width: ${theme.breakpoints.sm}px) and (max-width: ${theme
              .breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'lg':
        return css`
          @media (min-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'to-sm':
        return css`
          @media (max-width: ${theme.breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'to-md':
        return css`
          @media (max-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'to-lg':
        return css`
          @media (max-width: ${theme.breakpoints.lg}px) {
            ${css(...args)};
          }
        `;

      case 'from-xs':
        return css`
          @media (min-width: ${theme.breakpoints.xs}px) {
            ${css(...args)};
          }
        `;

      case 'from-sm':
        return css`
          @media (min-width: ${theme.breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'from-md':
        return css`
          @media (min-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'from-lg':
        return css`
          @media (min-width: ${theme.breakpoints.lg}px) {
            ${css(...args)};
          }
        `;

      case 'from-xlg':
        return css`
          @media (min-width: ${theme.breakpoints.xlg}px) {
            ${css(...args)};
          }
        `;
    }
  };
}
