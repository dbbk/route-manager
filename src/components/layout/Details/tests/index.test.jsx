// @flow

import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { Details } from '@festicket/react-ui-components';

import theme from 'src/utils/theme';
import { renderWithTheme } from 'src/tests/utils';
import { title, body } from './fixtures';

describe('<Details />', () => {
  test('with initial props', () => {
    const tree = renderWithTheme(<Details title={title}>{body}</Details>);
    expect(tree).toMatchSnapshot();
  });

  test('with openInitially prop', () => {
    const tree = renderWithTheme(
      <Details title={title} openInitially>
        {body}
      </Details>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('should open on button click', () => {
    const tree = mount(
      <ThemeProvider theme={theme}>
        <Details title={title}>{body}</Details>
      </ThemeProvider>,
    );

    const button = tree.find('button');

    expect(tree.find('BodyWrapper').prop('open')).toBe(false);

    button.simulate('click');

    expect(tree.find('BodyWrapper').prop('open')).toBe(true);
  });
});
