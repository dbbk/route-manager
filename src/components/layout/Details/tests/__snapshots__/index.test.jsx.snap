// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> with initial props 1`] = `
.c3 {
  border: none;
  outline: none;
}

.c5 {
  height: 18px;
  fill: #A2A4AD;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 28px;
  height: 28px;
}

.c2 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  padding: 0;
  cursor: pointer;
  background-color: #F5F5F5;
}

.c2:hover {
  background-color: rgba(36,37,41,0.1);
  color: #9497a1;
}

.c1 {
  font-weight: 700;
  color: #242529;
  margin-bottom: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 18px;
  line-height: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.c6 {
  display: none;
}

@media (min-width:568px) {
  .c1 {
    font-size: 20px;
    line-height: 22px;
  }
}

<div>
  <div
    className="c0"
    onClick={[Function]}
    open={false}
  >
    <h3
      className="c1"
      id={undefined}
    >
      Lorem ipsum dolor sit amet
    </h3>
    <button
      aria-disabled={undefined}
      className="c2 c3"
      disabled={undefined}
      size="regular"
    >
      <div
        className="c4"
      >
        <svg
          className="c5"
          color="grey"
          viewBox="0 0 18 18"
        >
          <path
            d="M14.7 5.8L9 11.4 3.3 5.8 1.9 7.2l7.1 7 7.1-7z"
          />
        </svg>
      </div>
    </button>
  </div>
  <div
    className="c6"
    open={false}
  >
    Et aeterno impedit accusamus nec. Ne porro altera recusabo eum, eos ad
movet mentitum, id malorum expetenda ius. Nec illud melius repudiandae
et. Quas prima congue vis ne, erat malorum detraxit in nam. Regione
iuvaret consulatu duo cu.
  </div>
</div>
`;

exports[`<Details /> with openInitially prop 1`] = `
.c3 {
  border: none;
  outline: none;
}

.c5 {
  height: 18px;
  fill: #A2A4AD;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 28px;
  height: 28px;
}

.c2 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  padding: 0;
  cursor: pointer;
  background-color: #F5F5F5;
}

.c2:hover {
  background-color: rgba(36,37,41,0.1);
  color: #9497a1;
}

.c1 {
  font-weight: 700;
  color: #242529;
  margin-bottom: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 18px;
  line-height: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
  margin-bottom: 20px;
}

.c6 {
  display: block;
}

@media (min-width:568px) {
  .c1 {
    font-size: 20px;
    line-height: 22px;
  }
}

<div>
  <div
    className="c0"
    onClick={[Function]}
    open={true}
  >
    <h3
      className="c1"
      id={undefined}
    >
      Lorem ipsum dolor sit amet
    </h3>
    <button
      aria-disabled={undefined}
      className="c2 c3"
      disabled={undefined}
      size="regular"
    >
      <div
        className="c4"
      >
        <svg
          className="c5"
          color="grey"
          viewBox="0 0 18 18"
        >
          <path
            d="M3.3 13.2L9 7.6l5.7 5.6 1.4-1.4-7.1-7-7.1 7z"
          />
        </svg>
      </div>
    </button>
  </div>
  <div
    className="c6"
    open={true}
  >
    Et aeterno impedit accusamus nec. Ne porro altera recusabo eum, eos ad
movet mentitum, id malorum expetenda ius. Nec illud melius repudiandae
et. Quas prima congue vis ne, erat malorum detraxit in nam. Regione
iuvaret consulatu duo cu.
  </div>
</div>
`;
